cmd/machine-config-operator/start.go

```diff
@@ -91,20 +98,24 @@ func runStartCmd(cmd *cobra.Command, args []string) {
 
 		go controller.Run(2, ctrlctx.Stop)
 
-		select {}
+		// wait here in this function until the context gets cancelled (which tells us whe were being shut down)
+		<-ctx.Done()
+		glog.Infof("Context has returned done, waiting 10 seconds to let any in-progress work finish")
+		<-time.After(10 * time.Second)
```

Comment:
Perhaps worth factoring this into a helper function to share with controller?

Or at least just a cookie for future readers to know there's code duplication, like
`// Note, if you're changing this you probably also want to change machine-config-controller/start.go`
or so?